### ISSUE TITLE
mbed-greentea 1.4.0 failed on Japanese Windows 10

### ISSUE THREAD

**Issue by toyowata** [ARMmbed/greentea#288]
Our partner Seeed reported this and I also reproduced the issue.
cc @ytsuboi

Problem:
`mbed test` failed by UnicodeDecodeError on Windows 10.

[python version]
Python 2.7.15

[mbed-os version]
mbed-os (#bf6f2c3c6434, tags: latest, mbed-os-5.10.3)

[Windows version]
Microsoft Windows [Version 10.0.17134.345]

Steps to reproduce:
* Import mbed-os-5.10.3
* Install missing python modules
* Run mbed test (any target, any toolchain)

```
PS C:\test\mbed-os> mbed test -m DISCO_L475VG_IOT01A -t gcc_arm -n tests-mbed_hal-ticker --run
[mbed] Auto-installing missing Python modules...
mbedgt: greentea test automation tool ver. 1.4.0
mbedgt: test specification file '.\BUILD\tests\DISCO_L475VG_IOT01A\GCC_ARM\test_spec.json' (specified with --test-spec option)
mbedgt: using '.\BUILD\tests\DISCO_L475VG_IOT01A\GCC_ARM\test_spec.json' from current directory!
mbedgt: detecting connected mbed-enabled devices...
mbedgt: detected 1 device
mbedgt: processing target 'DISCO_L475VG_IOT01A' toolchain 'GCC_ARM' compatible platforms... (note: switch set to --parallel 1)
mbedgt: test case filter (specified with -n option)
        tests-mbed_hal-ticker
        test filtered in 'tests-mbed_hal-ticker'
mbedgt: running 1 test for platform 'DISCO_L475VG_IOT01A' and toolchain 'GCC_ARM'
mbedgt: mbed-host-test-runner: started
Exception in thread Thread-3:
Traceback (most recent call last):
  File "c:\python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "c:\python27\lib\threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "c:\python27\lib\site-packages\mbed_greentea\mbed_greentea_cli.py", line 489, in run_test_thread
    verbose=verbose)
  File "c:\python27\lib\site-packages\mbed_greentea\mbed_test_api.py", line 318, in run_host_test
    returncode, htrun_output = run_htrun(cmd, verbose)
  File "c:\python27\lib\site-packages\mbed_greentea\mbed_test_api.py", line 134, in run_htrun
    htrun_output += line.decode('utf-8')
  File "c:\python27\lib\encodings\utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0x8c in position 10: invalid start byte

mbedgt: could not generate test report
mbedgt: completed in 6.17 sec
mbedgt: exited with code -1000
[mbed] ERROR: "mbedgt" returned error.
       Code: -1000
       Path: "C:\test\mbed-os"
       Command: "mbedgt --test-spec .\BUILD\tests\DISCO_L475VG_IOT01A\GCC_ARM\test_spec.json -n tests-mbed_hal-ticker"
       Tip: You could retry the last command with "-v" flag for verbose output
---
```

The mbed-greentea 1.3.3 does *not* fail the test.

More detail log (with -vv) is as below:
[mbed-geentea-140.txt](https://github.com/ARMmbed/greentea/files/2568115/mbed-geentea-140.txt)
[mbed-geentea-133.txt](https://github.com/ARMmbed/greentea/files/2568116/mbed-geentea-133.txt)

**Comment by bridadan** [comment]
Hi @toyowata, I believe I have a fix in #289. Would it be possible to try that branch and see if that addresses the issue?

**Comment by toyowata** [comment]
Hi @bridadan 

Your patch fixed the issue as below.  Thanks a lot!

```
PS C:\test\mbed-os> mbed test -m DISCO_L475VG_IOT01A -t gcc_arm -n tests-mbed_hal-ticker --run
[mbed] Auto-installing missing Python modules...
mbedgt: greentea test automation tool ver. 1.4.0
mbedgt: test specification file '.\BUILD\tests\DISCO_L475VG_IOT01A\GCC_ARM\test_spec.json' (specified with --test-spec option)
mbedgt: using '.\BUILD\tests\DISCO_L475VG_IOT01A\GCC_ARM\test_spec.json' from current directory!
mbedgt: detecting connected mbed-enabled devices...
mbedgt: detected 1 device
mbedgt: processing target 'DISCO_L475VG_IOT01A' toolchain 'GCC_ARM' compatible platforms... (note: switch set to --parallel 1)
mbedgt: test case filter (specified with -n option)
        tests-mbed_hal-ticker
        test filtered in 'tests-mbed_hal-ticker'
mbedgt: running 1 test for platform 'DISCO_L475VG_IOT01A' and toolchain 'GCC_ARM'
mbedgt: mbed-host-test-runner: started
mbedgt: checking for GCOV data...
mbedgt: test on hardware with target id: 076402216E1E6567515AF33E
mbedgt: test suite 'tests-mbed_hal-ticker' ........................................................... OK in 25.96 sec
        test case: 'legacy insert event head' ........................................................ OK in 0.06 sec
        test case: 'legacy insert event in overflow range' ........................................... OK in 0.07 sec
        test case: 'legacy insert event multiple overflow' ........................................... OK in 0.07 sec
        test case: 'legacy insert event outside overflow range' ...................................... OK in 0.07 sec
        test case: 'legacy insert event overflow' .................................................... OK in 0.06 sec
        test case: 'legacy insert event tail' ........................................................ OK in 0.06 sec
        test case: 'test_frequencies_and_masks' ...................................................... OK in 1.73 sec
        test case: 'test_insert_event_us_head' ....................................................... OK in 0.06 sec
        test case: 'test_insert_event_us_in_overflow_range' .......................................... OK in 0.07 sec
        test case: 'test_insert_event_us_multiple_random' ............................................ OK in 0.06 sec
        test case: 'test_insert_event_us_outside_overflow_range' ..................................... OK in 0.08 sec
        test case: 'test_insert_event_us_tail' ....................................................... OK in 0.05 sec
        test case: 'test_insert_event_us_underflow' .................................................. OK in 0.07 sec
        test case: 'test_irq_handler_insert_immediate_in_irq' ........................................ OK in 0.07 sec
        test case: 'test_irq_handler_insert_non_immediate_in_irq' .................................... OK in 0.07 sec
        test case: 'test_irq_handler_multiple_event_multiple_dequeue' ................................ OK in 0.08 sec
        test case: 'test_irq_handler_multiple_event_single_dequeue' .................................. OK in 0.07 sec
        test case: 'test_irq_handler_multiple_event_single_dequeue_overflow' ......................... OK in 0.09 sec
        test case: 'test_irq_handler_single_event' ................................................... OK in 0.07 sec
        test case: 'test_irq_handler_single_event_spurious' .......................................... OK in 0.08 sec
        test case: 'test_legacy_insert_event_multiple_random' ........................................ OK in 0.07 sec
        test case: 'test_match_interval_passed' ...................................................... OK in 0.06 sec
        test case: 'test_match_interval_passed_table' ................................................ OK in 0.07 sec
        test case: 'test_remove_event_head' .......................................................... OK in 0.04 sec
        test case: 'test_remove_event_invalid' ....................................................... OK in 0.06 sec
        test case: 'test_remove_event_tail' .......................................................... OK in 0.05 sec
        test case: 'test_remove_random' .............................................................. OK in 0.05 sec
        test case: 'test_set_interrupt_past_time' .................................................... OK in 0.06 sec
        test case: 'test_set_interrupt_past_time_with_delay' ......................................... OK in 0.07 sec
        test case: 'test_suspend_resume' ............................................................. OK in 0.05 sec
        test case: 'test_ticker_max_value' ........................................................... OK in 0.06 sec
        test case: 'ticker initialization' ........................................................... OK in 0.06 sec
        test case: 'ticker multiple initialization' .................................................. OK in 0.06 sec
        test case: 'ticker read' ..................................................................... OK in 0.05 sec
        test case: 'ticker read overflow' ............................................................ OK in 0.04 sec
        test case: 'update overflow guard' ........................................................... OK in 0.05 sec
        test case: 'update overflow guard in case of spurious interrupt' ............................. OK in 0.07 sec
mbedgt: test case summary: 37 passes, 0 failures
mbedgt: all tests finished!
mbedgt: shuffle seed: 0.9260426916
mbedgt: test suite report:
| target                      | platform_name       | test suite            | result | elapsed_time (sec) | copy_method |
|-----------------------------|---------------------|-----------------------|--------|--------------------|-------------|
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | OK     | 25.96              | default     |
mbedgt: test suite results: 1 OK
mbedgt: test case report:
| target                      | platform_name       | test suite            | test case                                               | passed | failed | result | elapsed_time (sec) |
|-----------------------------|---------------------|-----------------------|---------------------------------------------------------|--------|--------|--------|--------------------|
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | legacy insert event head                                | 1      | 0      | OK     | 0.06               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | legacy insert event in overflow range                   | 1      | 0      | OK     | 0.07               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | legacy insert event multiple overflow                   | 1      | 0      | OK     | 0.07               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | legacy insert event outside overflow range              | 1      | 0      | OK     | 0.07               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | legacy insert event overflow                            | 1      | 0      | OK     | 0.06               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | legacy insert event tail                                | 1      | 0      | OK     | 0.06               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_frequencies_and_masks                              | 1      | 0      | OK     | 1.73               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_insert_event_us_head                               | 1      | 0      | OK     | 0.06               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_insert_event_us_in_overflow_range                  | 1      | 0      | OK     | 0.07               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_insert_event_us_multiple_random                    | 1      | 0      | OK     | 0.06               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_insert_event_us_outside_overflow_range             | 1      | 0      | OK     | 0.08               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_insert_event_us_tail                               | 1      | 0      | OK     | 0.05               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_insert_event_us_underflow                          | 1      | 0      | OK     | 0.07               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_irq_handler_insert_immediate_in_irq                | 1      | 0      | OK     | 0.07               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_irq_handler_insert_non_immediate_in_irq            | 1      | 0      | OK     | 0.07               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_irq_handler_multiple_event_multiple_dequeue        | 1      | 0      | OK     | 0.08               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_irq_handler_multiple_event_single_dequeue          | 1      | 0      | OK     | 0.07               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_irq_handler_multiple_event_single_dequeue_overflow | 1      | 0      | OK     | 0.09               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_irq_handler_single_event                           | 1      | 0      | OK     | 0.07               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_irq_handler_single_event_spurious                  | 1      | 0      | OK     | 0.08               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_legacy_insert_event_multiple_random                | 1      | 0      | OK     | 0.07               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_match_interval_passed                              | 1      | 0      | OK     | 0.06               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_match_interval_passed_table                        | 1      | 0      | OK     | 0.07               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_remove_event_head                                  | 1      | 0      | OK     | 0.04               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_remove_event_invalid                               | 1      | 0      | OK     | 0.06               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_remove_event_tail                                  | 1      | 0      | OK     | 0.05               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_remove_random                                      | 1      | 0      | OK     | 0.05               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_set_interrupt_past_time                            | 1      | 0      | OK     | 0.06               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_set_interrupt_past_time_with_delay                 | 1      | 0      | OK     | 0.07               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_suspend_resume                                     | 1      | 0      | OK     | 0.05               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | test_ticker_max_value                                   | 1      | 0      | OK     | 0.06               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | ticker initialization                                   | 1      | 0      | OK     | 0.06               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | ticker multiple initialization                          | 1      | 0      | OK     | 0.06               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | ticker read                                             | 1      | 0      | OK     | 0.05               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | ticker read overflow                                    | 1      | 0      | OK     | 0.04               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | update overflow guard                                   | 1      | 0      | OK     | 0.05               |
| DISCO_L475VG_IOT01A-GCC_ARM | DISCO_L475VG_IOT01A | tests-mbed_hal-ticker | update overflow guard in case of spurious interrupt     | 1      | 0      | OK     | 0.07               |
mbedgt: test case results: 37 OK
mbedgt: completed in 28.52 sec
PS C:\test\mbed-os>

```
